crates/sui-node/src/lib.rs

```diff
@@ -468,6 +458,9 @@ impl SuiNode {
         let cache_metrics = Arc::new(ResolverMetrics::new(&prometheus_registry));
         let signature_verifier_metrics = SignatureVerifierMetrics::new(&prometheus_registry);
 
+        let cache_traits =
+            build_execution_cache(&epoch_start_configuration, &prometheus_registry, &store);
```

Comment:
I'm not sure if I read the code correctly. First time learning how to use `epoch_start_configuration`.

My question is that it appears `build_execution_cache` is still not guaranteed to be called during epoch change.

When updating node config to switch cache implementation, during epoch change, it seems will only change the `epoch_start_configuration` stored in DB, but won't actually switch the underlying cache implementation.

To actually switch the cache implementation, we need to restart the node, then read `epoch_start_configuration` to change cache implementation, which may still be in the middle of the epoch?

Reply:
> it seems will only change the epoch_start_configuration stored in DB, but won't actually switch the underlying cache implementation.

this is intended - it writes the cache impl for the next epoch to the epoch start config, and then that flag takes effect at the beginning of the next epoch. `epoch_start_config` is immutable within the epoch, so as long as the cache impl is a pure function of the value of `epoch_start_config` (it is), this should be correct.

Reply:
ok, fixed this to work properly now, thanks for spotting this